shed/event_streams.py

```diff
@@ -742,17 +760,40 @@ def __init__(self, predicate, child, *args, input_info,
         self.truth_value = None
         # Note we don't override event because with this update we don't see it
         if document_name == 'start':
-            self.update = self._update
+            self.update = self._start_update
+        elif document_name == 'descriptor':
+            self.update = self._descriptor_update
+            self.descriptor_truth_values = {}
         self.generate_provenance(predicate=predicate)
 
-    def _update(self, x, who=None):
+    def _start_update(self, x, who=None):
         # TODO: should we have something like event_contents for starts?
         name, docs = self.curate_streams(x, False)
         if name == 'start':
             self.truth_value = self.predicate(docs)
         if self.truth_value:
             self.emit((name, docs))
 
+    def _descriptor_update(self, x, who=None):
+        name, docs = self.curate_streams(x, False)
+        ret = None
+        if name == 'start':
+            self.descriptor_truth_values = {}
+            ret = super().start(docs)
+        elif name == 'descriptor':
+            # TODO: we need to sort out how to deal with nodes/modes which only
+            # take in singular headers vs multiple
+            self.descriptor_truth_values[docs[0]['uid']] = self.predicate(docs)
+            if self.descriptor_truth_values[docs[0]['uid']]:
+                ret = super().descriptor(docs)
+        elif (name == 'event' and
+              self.descriptor_truth_values[docs[0]['descriptor']]):
+            ret = super().event(docs)
+        elif name == 'stop':
```

Comment:
should you clear the `descriptor_truth_values` here?

Reply:
They are cleared on start for ease of debugging.